src/main/scala/zio/s3/Live.scala

```diff
@@ -105,76 +108,71 @@ final class Live(unsafeClient: S3AsyncClient) extends S3.Service {
   def multipartUpload[R <: zio.Has[_]: Tag](
     bucketName: String,
     key: String,
-    contentType: String,
     content: ZStream[R, Throwable, Byte],
-    metadata: Map[String, String] = Map.empty
+    options: MultipartUploadOptions = MultipartUploadOptions.default
   ): ZIO[R, S3Exception, Unit] =
     for {
-      uploadId <- execute(
-                    _.createMultipartUpload(
-                      CreateMultipartUploadRequest
-                        .builder()
-                        .bucket(bucketName)
-                        .key(key)
-                        .contentType(contentType)
-                        .metadata(metadata.asJava)
-                        .build()
-                    )
-                  ).map(_.uploadId())
-
-      parts    <- content
-                  //part size limit is 5Mb, required by amazon api
-                    .chunkN(5 * 1024 * 1024)
-                    .mapChunks(Chunk.single)
-                    .zipWithIndex
-                    .mapM {
-                      case (chunk, partNumber) =>
-                        execute(
-                          _.uploadPart(
-                            UploadPartRequest
-                              .builder()
-                              .bucket(bucketName)
-                              .key(key)
-                              .partNumber(partNumber.toInt + 1)
-                              .uploadId(uploadId)
-                              .contentLength(chunk.length.toLong)
-                              .build(),
-                            AsyncRequestBody.fromBytes(chunk.toArray)
-                          )
-                        ).map(r => CompletedPart.builder().partNumber(partNumber.toInt + 1).eTag(r.eTag()).build())
-                    }
-                    .runCollect
-                    .mapError(S3ExceptionUtils.fromThrowable)
-
-      _        <- execute(
-                    _.completeMultipartUpload(
-                      CompleteMultipartUploadRequest
-                        .builder()
-                        .bucket(bucketName)
-                        .key(key)
-                        .multipartUpload(CompletedMultipartUpload.builder().parts(parts.asJavaCollection).build())
-                        .uploadId(uploadId)
-                        .build()
-                    )
-                  )
+      uploadId              <- execute(
+                                 _.createMultipartUpload(
+                                   MultipartUploadOptions
+                                     .multipartUploadBuilder(options)
+                                     .bucket(bucketName)
+                                     .key(key)
+                                     .build()
+                                 )
+                               ).map(_.uploadId())
+
+      chunkedStream          = content.chunkN(options.partSize).mapChunks(Chunk.single).mapError(S3ExceptionUtils.fromThrowable)
```

Comment:
i  better like this approach, but i was wondering if my solution was working with amazon s3, if not your solution is totaly valid, was not aware of peel. good to know :)

Reply:
oki i tested my solution locally, and amazon s3 don't like it.

Reply:
https://github.com/zio/zio-s3/pull/117/commits/a1067926d148b8d2bf79a520fa5771945a73510c